src/index.test.ts

```diff
@@ -3788,176 +3791,2407 @@ describe('personalSign', function () {
-  );
-  expect(signature).toBe(
-    '0x49e75d475d767de7fcc67f521e0d86590723d872e6111e51c393e8c1e2f21d032dfaf5833af158915f035db6af4f37bf2d5d29781cd81f28a44c5cb4b9d241531b',
-  );
+    it('should throw when given an empty array', function () {
+      expect(() =>
+        sigUtil.signTypedData(
+          privateKey,
+          {
+            data: [],
+          },
+          'V1',
+        ),
+      ).toThrow('Expect argument to be non-empty array');
+    });
 
-  expect(address).toBe(recovered);
-});
+    describe('example data', function () {
+      // Reassigned to silence "no-loop-func" ESLint rule
+      // It was complaining because it saw that `it` and `expect` as "modified variables from the outer scope"
+      // which can be dangerous to reference in a loop. But they aren't modified in this case, just invoked.
+      const _expect = expect;
+      const _it = it;
 
-it('signTypedData and recoverTypedSignature V1 - multiple messages', function () {
-  const address = '0x29c76e6ad8f28bb1004902578fb108c507be341b';
-  const privKeyHex =
-    '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0';
+      for (const type of allSignTypedDataV1ExampleTypes) {
+        describe(`type "${type}"`, function () {
+          // Test all examples that do not crash
+          const inputs = signTypedDataV1Examples[type] || [];
+          for (const input of inputs) {
+            const inputType = input instanceof Buffer ? 'Buffer' : typeof input;
+            _it(`should sign "${input}" (type "${inputType}")`, function () {
+              _expect(
+                sigUtil.signTypedData(
+                  privateKey,
+                  {
+                    data: [{ name: 'data', type, value: input }],
+                  },
+                  'V1',
+                ),
+              ).toMatchSnapshot();
+            });
+          }
 
-  const privKey = Buffer.from(privKeyHex, 'hex');
+          // Test all examples that crash
+          const errorInputs = signTypedDataV1ErrorExamples[type] || [];
+          for (const { input, errorMessage } of errorInputs) {
+            const inputType = input instanceof Buffer ? 'Buffer' : typeof input;
+            _it(
+              `should fail to sign "${input}" (type "${inputType}")`,
+              function () {
+                _expect(() =>
+                  sigUtil.signTypedData(
+                    privateKey,
+                    {
+                      data: [{ name: 'data', type, value: input }],
+                    },
+                    'V1',
+                  ),
+                ).toThrow(errorMessage);
+              },
+            );
+          }
 
-  const typedData = [
-    {
-      type: 'string',
-      name: 'message',
-      value: 'Hi, Alice!',
-    },
-    {
-      type: 'uint8',
-      name: 'value',
-      value: 10,
-    },
-  ];
+          if (type === 'bytes') {
+            _it(
+              `should fail to sign array of all ${type} example data`,
```

Comment:
Fifth highlight - V1 supports arrays of any type except `bytes`. Apparently an array of `bytes` is unacceptable.

Reply:
Anything can be in an array though. From the page you just linked:

>Array elements can be of any type, including mapping or struct.

So this is an array of `bytes`. An array of an array of bytes really.